docs/detections/session-view.asciidoc

```diff
@@ -84,4 +90,40 @@ Session View includes an additional badge not pictured above:
 * The *Exec user change* badge highlights exec user changes, such as when a user escalates to root:
 +
 [role="screenshot"]
-image::images/session-view-exec-user-change-badge.png[The Exec user change button,width=80%,height=80%]
+image::images/session-view-exec-user-change-badge.png[The Exec user change badge,width=80%,height=80%]
+
+* The *Output* badge appears next to commands that generated terminal output. Click it to open the terminal output view.
++
+[role="screenshot"]
+image::images/session-view-output-badge.png[The Output badge,width=80%,height=80%]
+
+[[session-view-output]]
+[discrete]
+=== Terminal output view UI
+
+beta::[]
+
+Terminal output view provide a more complete understanding of commands executed by users or adversaries, and assists with auditing and compliance. It allows you to review a session's terminal output in the context of the original commands. It includes both terminal output and shell commands (if the shell was in echo mode).
+
+To enable terminal output data capture:
+
+. Go to *Manage* -> *Policies*, then select one or more of your {elastic-defend} integration policies to edit.
+. On the *Policy settings* tab, scroll down to the Linux event collection section near the bottom of the page
+and select the *Include session data* and *Capture terminal output* options.
+
+You can configure several additional settings by clicking *Advanced settings* at the bottom of the page:
+
+* `linux.advanced.tty_io.max_kilobytes_per_process`: The maximum number of kilobytes of output to record from a single process. Default: 512 KB. Process output exceeding this value will not be recorded.
+* `linux.advanced.tty_io.max_kilobytes_per_event`: The maximum number of kilobytes of output to send to {es} as a single event. Default: 512 KB. Additional data is captured as a new event.
+* `linux.advanced.tty_io.max_event_interval_seconds`: The maximum interval (in seconds) during which output is batched. Default: 30 seconds. Output will be sent to {es} at this interval (unless it first exceeds the `max_kilobytes_per_event` value, in which case it might be sent sooner).
+
+[role="screenshot"]
```

Comment:
Do we have any docs on writing alerting rules based on matching on terminal output?    

It is important to note that when writing Alerting Rules that query Elasticsearch for matching terminal output that multi-line matching will not behave consistently because the output text may be stored in more than one output event.  For example,  Foo\nBar - that is, `Foo` on one line followed by `Bar` on the next line - will not behave consistently because Bar might be in output event after Foo.   Elastic tries to ensure a given line of terminal output is contained in exactly one output event to ensure in-line matches work as expected (beware of terminal color codes).   However, very long lines may be split across more than one output event.

Reply:
We do not have any docs specifically about alerting rules based on terminal output, yet. But we could mention something here about the issue you outlined. 

Any thoughts on how to work around that limitation? I guess one thing would be to increase the maximum size of output events to reduce the number of unexpected newlines, but that would still be inconsistent.

Reply:
Yes, increasing the max event buffering size and 30 second capture to be larger (60s?) create bigger events that are more likely to work for multi-line matches but there are still chances for inconsistency.   Also, IIRC we flush the output immediately if the program writing to the terminal changes - e.g. I ctrl-Z a program that is producing output and the shell then echos a prompt (the shell is another program writing to the terminal, causing a flush), then put the original program back into the foreground with `fg`.   THis will break up the output of the original program over at least 2 events.

If this important for some customers, the best way is to post process these output events to group all the output events of a given process, and while they're at it ideally scrub out terminal control codes for color, etc.

Reply:
As for rules that match against process.io.text, it is advised that all rules match with leading/trailing wildcards (*).

e.g
```process.io.text: *hello world*```

EQL equivalent:
```process where process.io.text : "*hello world*"```

This will ensure a less strict rule which will have a much better chance of firing. Without the * wildcards, it would try to match an event with exactly the string "hello world" which in most cases will not match on anything, since an output event can have multiple lines in it. This also opens up another potential bit of confusion for the end user. E.g if their event size is quite large. e.g 512kb (due to some long cat'd file), there will be a single alert for this document no matter how little the text we are matching against. This could make it difficult to figure out which part of the proces.io.text value actually triggered the alert. We may add some UX down the road to help with this. e.g showing alerts where event.action: 'text_output' inside the tty player itself (potentially highlighting the bit of text responsible for the alert). Reducing the timeout and batch size configs could help limit the amount of text per event, thus alleviating this issue.